compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaPathParameterSymbol.java

```diff
@@ -105,8 +115,15 @@ public TypeSymbol typeDescriptor() {
             return this.typeDescriptor;
         }
 
+        BType bType;
+        if (this.segmentKind == Kind.PATH_REST_PARAMETER && this.getInternalSymbol() instanceof BVarSymbol) {
```

Comment:
Will the latter check ever be true now?

Reply:
We need it due to this https://github.com/ballerina-platform/ballerina-lang/pull/39477#discussion_r1299091767